### ISSUE TITLE
Ride and the interpreter have different opinions on how many windows are open

### ISSUE THREAD

**Issue by aplteam** [Dyalog/ride#557]
### Describe the issue you are having

I was constantly tracing and editing (different) functions when all of a sudden `<Ctl+Enter>` dit nor open the Tracer window any more. As it turned out `)ed {functionname}` had no effect either. Everything else worked fine.

Apparently the interpreter believed that there was still a window open on that function while Ride was not. I have no idea how I got into this state.

I investigated the issue with Gil and we realized that there are two issues here:

1. The log was of limited use because the number of rows it keeps track of is too small, at least when the status bar is viewed.

2. There should be a mechanism that allows the interpreter and Ride to re-sync on the number of windows currently open.

In my case the "Close all Windows" menu command solved the issue.

### Did you connect to an already running interpreter or start the interpreter from RIDE?

n/a

### How do you reproduce the issue?

Unfortunately not at all.

### Paste the contents of Help → About (Shift+F1)
```
IDE:
  Version: 4.2.3432
  Platform: Linux x86_64
  Date: 2019-05-24 12:10:06 +0100
  Git commit: 3be9ea797d360ddb13d87752f36752316f1e19fa
  Preferences:{
    "autoPW":"1",
    "editWins":"{\"width\":1181,\"height\":854,\"x\":735,\"y\":141,\"ox\":0,\"oy\":0}",
    "editWinsRememberPos":"1",
    "floatSingle":"0",
    "floating":"1",
    "kbdLocale":"en_GB",
    "minimapEnabled":"1",
    "persistentHistorySize":"300",
    "squiggleTips":"1",
    "keys":"{\"JSC\":[],\"LOG\":[\"Ctrl-F12\",\"F12\"]}",
    "sbar":"0"
  }

Interpreter:
  Version: unknown
  Platform: unknown
  Edition: unknown
  Date: unknown
```

**Comment by aplteam** [comment]
Continue to work and it happened again when I was actually working _slowly_. My original assumption that something got of sync because I caused timing issues by working too fast was wrong.

**Comment by e9gille** [comment]
Probably fixed by #656 
Feel free to reopen if it happens again.
